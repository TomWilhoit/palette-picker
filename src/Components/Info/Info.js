import React, { Component } from "react";
import { InstructionBox } from "../InstructionBox/InstructionBox"

export class Info extends Component { 

  handleClick = (e) => {
    var { className } = e.target
    if (className.includes("close")) {
      this.props.hideInfo(e);
    }
    if (className.includes("instruction-step")) {
        let infoType = className.split(" ").find(word => !word.includes("instruction-step"));
        let infoElement = document.querySelector(`#${infoType}`)
        infoElement.classList.toggle('show-info')
    }
  }

  render() {
    return (
        <div className="info-container">
          <section>
            <header>
              <h1>Instructions</h1>
              <div 
                className="close-info-btn close" 
                onClick={(e) => this.handleClick(e)}
              >
                <i className="fas fa-times close"></i>
              </div>
              <h4>Thank you for checking out Palette Picker!</h4>
            </header>
            <main>
              <p>
                Create/select Projects and Palettes on the right.
                Mix/name/save Palettes on the bottom.
                Check detailed instructions by clicking the steps below:
              </p>
              <div 
                className="instruction-step project-step"
                onClick={(e) => this.handleClick(e)}
              >
                1. Create/Select Project 
                <div className="project-info detail-box" id="project-step">
                  <InstructionBox
                    message="First, create or select a project. On the top right area of the console, you can type a project name in the input and click the green plus button next to it to create a new project. Below this input in a scroll box, you can click newly created or previously saved projects to select them."
                  />
                </div>
              </div>
              <div 
                className="instruction-step palette-step"
                onClick={(e) => this.handleClick(e)}
              >
                2. Create/Select Palette
                <div className="palette-info detail-box" id="palette-step">
                  <InstructionBox
                    message="Next, choose whether to create a palette or select a saved palette. On the lower right area of the console, either click a previously saved palette in the scroll box, or the bar above that says 'Create New Palette', highlighted by default." 
                    id="palette-info"
                  />
                </div>
              </div>
              <div 
                className="instruction-step mix-step"
                onClick={(e) => this.handleClick(e)}
              >
                3. Make a Color Palette!
                <div className="mix-info detail-box" id="mix-step">
                  <InstructionBox
                    message=" Now, we are ready to make a palette! On the bottom of the screen, double check which project and Palette you have selected, unlock colors you'd like to replace and lock any colors you'd like to keep by toggling the locks each color, then click the 'Mix Palette!' button to randomize unlocked colors. New palettes start with all colors unlocked, but saved palettes start with colors locked. To save a palette, name it in the input on the bottom of the screen and click submit. Previously saved palettes do not need to be named again, but if a new name is entered in the input, the palette will be renamed upon submission."
                    id="mix-info"
                  />
                </div>
              </div>
            </main>
          </section>
        </div>
    )  
  }
}