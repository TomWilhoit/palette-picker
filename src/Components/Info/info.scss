.info-container {
  background-color: $light-bkgd;
  border: 4px solid $border;
  border-radius: 10px;
  box-shadow: 5px 8px 7px $shadow;
  padding: 10px 0 30px 0;
  h1 {
    margin: 14px 0;
    position: relative;
  }
  h4 {
    font-size: 120%;
    margin-bottom: 10px;
  }
  p {
    margin: 0 20px 10px 20px;
  }
}

.close-info-btn {
  @include delete-btn;
  position: absolute;
  right: 10px;
  top: 10px;
}

.instruction-step {
  @include icon-btn-hover;
  border: 2px solid $border;
  box-shadow: 0 3px 4px $shadow;
  margin: 10px 10% 0 10%;
  position: relative;
  width: 80%;
}

.detail-box {
  height: 0;
  opacity: 0;
  visibility: hidden;
}

.show-info {
  height: auto;
  opacity: 1;
  transition: all .75s ease-in-out;
  visibility: visible;
}

.arrow {
  display: inline-block;
  left: 78%;
  position: absolute;
  top: -1px;
}

.arrow-down {
  transform: rotate(90deg);
}

.fa-chevron-right {
  font-size: 10px;
}