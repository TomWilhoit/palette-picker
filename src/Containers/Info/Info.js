import React, { Component } from "react";
import { InstructionBox } from "../../Components/InstructionBox/InstructionBox"

export class Info extends Component { 

  handleClick = (e) => {
    var { className } = e.target;
    if (className.includes("close")) {
      this.props.hideInfo(e);
    }
    if (className.includes("instruction-step")) {
      this.toggleDetail(className); 
    }
    if (className.includes("instruction-box")) {
      let selectedElement = e.target.parentElement.parentElement.className;
      this.closeDetail(selectedElement);
    }
  }

  toggleDetail = (className) => {
    let infoType = className.split(" ").find(word => !word.includes("instruction-step"));
    let infoElement = document.querySelector(`#${infoType}`);
    let arrow = document.querySelector(`.${infoType}`).children[0];
    arrow.classList.toggle('arrow-down');
    infoElement.classList.toggle('show-info');
  }

  closeDetail = (selectedElement) => {
    let infoType = selectedElement.split(" ").find(word => !word.includes("instruction-step"));
    let infoElement = document.querySelector(`#${infoType}`);
    let arrow = document.querySelector(`.${infoType}`).children[0];
    arrow.classList.remove('arrow-down');
    infoElement.classList.remove('show-info');
  }

  render() {
    return (
      <div className="info-container">
        <section>
          <header>
            <h1>Instructions</h1>
            <div 
              className="close-info-btn close" 
              onClick={(e) => this.handleClick(e)}
            >
              <i className="fas fa-times close"></i>
            </div>
            <h4>Thank you for checking out Palette Picker!</h4>
          </header>
          <main>
            <p>
              Create or select Projects and Palettes on the right.
              Mix, name, or save Palettes on the bottom.
              Detailed instructions below:
            </p>
            <div 
              className="instruction-step project-step"
              onClick={(e) => this.handleClick(e)}
            >
              1. Create/Select Project <div className="arrow">></div>
              <div className="project-info detail-box" id="project-step" onClick={(e) => this.handleClick(e)}
>
                <InstructionBox
                  message="First, create or select a project. On the top right area of the console, you can type a project name in the input and click the green plus button next to it to create a new project. Below this input in a scroll box, you can click newly created or previously saved projects to select them."
                />
              </div>
            </div>
            <div 
              className="instruction-step palette-step"
              onClick={(e) => this.handleClick(e)}
            >
              2. Create/Select Palette <div className="arrow">></div>
              <div className="palette-info detail-box" id="palette-step"               onClick={(e) => this.handleClick(e)}
>
                <InstructionBox
                  message="Next, choose whether to create a palette or select a saved palette. On the lower right area of the console, either click a previously saved palette in the scroll box, or the bar above that says 'Create New Palette', highlighted by default." 
                />
              </div>
            </div>
            <div 
              className="instruction-step mix-step"
              onClick={(e) => this.handleClick(e)}
            >
              3. Make a Color Palette! <div className="arrow">></div>
              <div className="mix-info detail-box" id="mix-step"               onClick={(e) => this.handleClick(e)}
>
                <InstructionBox
                  message="Now, we are ready to make a palette! On the bottom of the screen, double check which project and Palette you have selected, unlock colors you'd like to replace and lock any colors you'd like to keep by toggling the locks each color, then click the 'Mix Palette!' button to randomize unlocked colors. New palettes start with all colors unlocked, but saved palettes start with colors locked. To save a palette, name it in the input on the bottom of the screen and click submit. Previously saved palettes do not need to be named again, but if a new name is entered in the input, the palette will be renamed upon submission."
                />
              </div>
            </div>
          </main>
        </section>
      </div>
    )  
  }
}