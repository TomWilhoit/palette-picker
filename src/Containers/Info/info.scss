.info-container {
  background-color: $background;
  border: 4px solid $border;
  border-radius: 10px;
  box-shadow: 5px 8px 7px $shadow;
  left: 29%;
  padding: 10px 0 30px 0;
  position: relative;
  top: 10%;
  width: 42%;
  z-index: 2;
  h1 {
    margin: 14px 0;
    position: relative;
  }
  h4 {
    font-size: 120%;
    margin-bottom: 10px;
  }
  p {
    margin: 0 20px 10px 20px;
  }
}

.close-info-btn {
  @include delete-btn;
  position: absolute;
  right: 10px;
  top: 10px;
}

.instruction-step {
  @include icon-btn-hover;
  border: 2px solid $border;
  box-shadow: 0 3px 4px $shadow;
  margin: 10px 10% 0 10%;
  width: 80%;
}

.detail-box {
  visibility: hidden;
  opacity: 0;
  height: 0;
}

.show-info {
  visibility: visible;
  height: auto;
  opacity: 1;
  transition: all .75s ease-in-out;
}

.arrow {
  display: inline-block;
  position: relative;
  left: 30px;
}

.arrow-down {
  transform: rotate(90deg);
}

.fa-chevron-right {
  font-size: 10px;
}