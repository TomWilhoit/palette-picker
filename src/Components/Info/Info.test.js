import React from "react";
import { Info } from "./Info";
import { shallow, mount } from "enzyme";

describe("Info", () => {
  let wrapper;
  let props;
  
  beforeEach(() => {
    props = { hideInfo: jest.fn() };
    wrapper = shallow(<Info {...props} />);
  })

  describe("on load", () => {
    it("should match the snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    })
  })

  describe("handleClick", () => {
    it("should activate when close-info-btn is pressed", () => {
      const e = { preventDefault: jest.fn() };
      wrapper.instance().handleClick = jest.fn();
      wrapper.find(".close-info-btn").first().simulate("click", e);
      expect(wrapper.instance().handleClick).toBeCalled();
    })

    it("should activate when an instruction-step is pressed", () => {
      const e = { preventDefault: jest.fn() };
      wrapper.instance().handleClick = jest.fn();
      wrapper.find(".instruction-step").first().simulate("click", e);
      expect(wrapper.instance().handleClick).toBeCalled();
    })

    it("should activate when a detail-box is pressed", () => {
      const e = { preventDefault: jest.fn() };
      wrapper.instance().handleClick = jest.fn();
      wrapper.find(".detail-box").first().simulate("click", e);
      expect(wrapper.instance().handleClick).toBeCalled();
    })

    it("should run hideInfo if 'closable' is in included the className", () => {
      const e = { preventDefault: jest.fn(), target: { className: "closable" }};
      wrapper.instance().handleClick(e);
      expect(wrapper.instance().props.hideInfo).toBeCalled();
    })

    it("should run toggleDetail if 'instruction-step' is included in the className", () => {
      const e = { preventDefault: jest.fn(), target: { className: "instruction-step" }};
      wrapper.instance().toggleDetail = jest.fn();
      wrapper.instance().handleClick(e);
      expect(wrapper.instance().toggleDetail).toBeCalled();
    })

    it("should run closeDetail if 'instruction-box' is included in the className", () => {
      const e = { preventDefault: jest.fn(), target: { className: "instruction-box", parentElement: { parentElement: { className: "instruction-step project-step" }}}};
      const selectedElement = jest.fn();
      wrapper.instance().closeDetail = jest.fn();
      wrapper.instance().handleClick(e);
      expect(wrapper.instance().closeDetail).toBeCalled();
    })
  })

  describe("toggleDetail", () => {
    //testing dom selectors/ class manipulation
    it.skip("should add 'show-info' to a class list if it is not there", () => {
      let mockClassName = "instruction-step project-step";
      expect(wrapper.find(".project-info").hasClass("show-info")).toEqual(false);
      wrapper.instance().toggleDetail(mockClassName);
      wrapper.instance().forceUpdate();
      expect(wrapper.find(".project-info").hasClass("show-info")).toEqual(true);
    })

    it.skip("should add 'arrow-down' to a class list if it is not there", () => {
      let mockClassName = "instruction-step project-step";
      expect(wrapper.find(".project-step").hasClass("arrow-down")).toEqual(false);
      wrapper.instance().toggleDetail(mockClassName);
      expect(wrapper.find(".project-step").hasClass("arrow-down")).toEqual(true);
    })

    it.skip("should remove 'show-info' to a class list if it is there already", () => {
      let mockClassName = "instruction-step project-step";
      wrapper.instÃŸance().toggleDetail(mockClassName);
      expect(wrapper.find("#project-step").hasClass("show-info")).toEqual(true);
      wrapper.instance().toggleDetail(mockClassName);
      expect(wrapper.find("#project-step").hasClass("show-info")).toEqual(false);
    })

    it.skip("should remove 'arrow-down' to a class list if it is there already", () => {
      let mockClassName = "instruction-step project-step";
      wrapper.instance().toggleDetail(mockClassName);
      expect(wrapper.find(".project-step").hasClass("arrow-down")).toEqual(true);
      wrapper.instance().toggleDetail(mockClassName);
      expect(wrapper.find(".project-step").hasClass("arrow-down")).toEqual(false);
    })
  })

  describe("closeDetail", () => {
    //testing dom selectors/ class manipulation
    it.skip("should remove 'show-info' from a detail", () => {
      let mockClassName = "instruction-step project-step";
      wrapper.instance().toggleDetail(mockClassName);
      expect(wrapper.find("#project-step").hasClass("show-info")).toEqual(true);
      wrapper.instance().closeDetail(mockClassName);
      expect(wrapper.find("#project-step").hasClass("show-info")).toEqual(false);
    })
    
    it.skip("should remove 'arrow-down' from a detail", () => {
      let mockClassName = "instruction-step project-step";
      wrapper.instance().toggleDetail(mockClassName);
      expect(wrapper.find(".project-step").hasClass("arrow-down")).toEqual(true);
      wrapper.instance().closeDetail(mockClassName);
      expect(wrapper.find(".project-step").hasClass("arrow-down")).toEqual(false);
    })
  })
})