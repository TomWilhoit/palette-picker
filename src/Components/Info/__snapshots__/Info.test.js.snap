// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info on load should match the snapshot 1`] = `
<div
  className="info-container"
>
  <header>
    <h1>
      Instructions
    </h1>
    <div
      className="close-info-btn closable"
      onClick={[Function]}
    >
      <i
        className="fas fa-times closable"
      />
    </div>
    <h4>
      Thank you for checking out Palette Picker!
    </h4>
  </header>
  <main>
    <p>
      Create or select Projects and Palettes on the right. Mix, name, or save Palettes on the bottom. Detailed instructions below:
    </p>
    <div
      className="instruction-step project-step"
      onClick={[Function]}
    >
      1. Create/Select Project
      <div
        className="arrow"
      >
        <i
          aria-hidden="true"
          className="fa fa-chevron-right"
        />
      </div>
      <div
        className="project-info detail-box"
        id="project-step"
        onClick={[Function]}
      >
        <InstructionBox
          message="First, create or select a project. On the top right area of the console, you can type a project name in the input and click the green plus button next to it to create a new project. Below this input in a scroll box, you can click newly created or previously saved projects to select them."
        />
      </div>
    </div>
    <div
      className="instruction-step palette-step"
      onClick={[Function]}
    >
      2. Create/Select Palette
      <div
        className="arrow"
      >
        <i
          aria-hidden="true"
          className="fa fa-chevron-right"
        />
      </div>
      <div
        className="palette-info detail-box"
        id="palette-step"
        onClick={[Function]}
      >
        <InstructionBox
          message="Next, choose whether to create a palette or select a saved palette. On the lower right area of the console, either click a previously saved palette in the scroll box, or the bar above that says 'Create New Palette', highlighted by default."
        />
      </div>
    </div>
    <div
      className="instruction-step mix-step"
      onClick={[Function]}
    >
      3. Make a Color Palette!
      <div
        className="arrow"
      >
        <i
          aria-hidden="true"
          className="fa fa-chevron-right"
        />
      </div>
      <div
        className="mix-info detail-box"
        id="mix-step"
        onClick={[Function]}
      >
        <InstructionBox
          message="Now, we are ready to make a palette! On the bottom of the screen, double check which project and Palette you have selected, unlock colors you'd like to replace and lock any colors you'd like to keep by toggling the locks each color, then click the 'Mix Palette!' button to randomize unlocked colors. New palettes start with all colors unlocked, but saved palettes start with colors locked. To save a palette, name it in the input on the bottom of the screen and click submit. Previously saved palettes do not need to be named again, but if a new name is entered in the input, the palette will be renamed upon submission."
        />
      </div>
    </div>
  </main>
</div>
`;
